src/Microsoft.DotNet.ProjectModel/RuntimeOutputFiles.cs

```diff
@@ -24,7 +24,15 @@ public string Executable
         {
             get
             {
-                var extension = FileNameSuffixes.CurrentPlatform.Exe;
+                string extension;
+                if (string.IsNullOrEmpty(_runtimeIdentifier))
+                {
+                    extension = FileNameSuffixes.CurrentPlatform.Exe;
+                }
+                else
+                {
+                    extension = _runtimeIdentifier.ToLower().StartsWith("win") ? ".exe" : string.Empty;
```

Comment:
we should create a common function for this, it's pretty specific, may potentially need to change, and is in multiple places